app/scripts/inpage.js

```diff
@@ -61,8 +61,20 @@ const inpageProvider = new MetamaskInpageProvider(metamaskStream)
 // set a high max listener count to avoid unnecesary warnings
 inpageProvider.setMaxListeners(100)
 
+let warnedOfAutoRefreshDeprecation = false
 // augment the provider with its enable method
 inpageProvider.enable = function ({ force } = {}) {
+  if (
+    !warnedOfAutoRefreshDeprecation &&
+    inpageProvider.autoRefreshOnNetworkChange &&
+    inpageProvider.listenerCount('networkChanged') === 0
```

Comment:
This one's an interesting decision. I almost think we should exclude this third criteria (`    inpageProvider.listenerCount('networkChanged') === 0`), because adding those listeners is basically useless without the `autoRefreshOnNetworkChange` param set, so while the listener indicates they are using a preparatory API, it doesn't mean they have actually seen it in action.

I could go either way though, it doesn't need to block.

Reply:
Hmmm 🤔 

Fair point - if we are encouraging anyone to implement the manual reload on network change, we should encourage them to set that flag as well so that they can actually test it reliably. I'll update this condition, and re-word the warning slightly.